client/src/store/types.ts

```diff
@@ -1,9 +1,15 @@
 import {Item} from "./item/types";
+import {User} from "./users/types";
 
 export type RootState = {
   item: ItemState;
 };
 
 export interface ItemState {
   items: Item[];
+}
+export interface UserState {
+  users: User[];
+  isLoggedIn: boolean;
```

Comment:
This boolean is the responsibility of the App state. The user state just describe the user who is currently logged in, and used for fetching data about them when they load different pages. The app state determines if someone is logged in, regardless of who the user is.